GetStream/stream-chat-swift
# Async Messages

### 🔗 Issue Links

Resolves https://github.com/GetStream/ios-issues-tracking/issues/295

### 🎯 Goal

The goal of this PR is to add the Voice Recording (Async Messages) feature on the LLC and UIKit frameworks.

### 📝 Summary

When the `Components.default.isVoiceRecordingEnabled` is set to `true`, you can record Voice messages and send them in your conversations. Additionally, you can playback Voice Recording messages that you have sent/received.

### 🛠 Implementation

The implementation is split in LLC and UIKit components. The main ones are:
In LLC
- StreamAudioPlayer
- StreamAudioQueuePlayer
- StreamAudioRecorder
- StreamAudioAnalysisEngine

In UIKit
- VoiceRecordingVC
- ChatMessageVoiceRecordingAttachmentListView+ItemView

### 🎨 Showcase

_Add relevant screenshots and/or videos/gifs to easily see what this PR changes, if applicable._

| Before | After |
| ------ | ----- |
| ![Simulator Screenshot - iPhone 14 Pro - 2023-05-24 at 17 59 55](https://github.com/GetStream/stream-chat-swift/assets/472467/ef902c8f-acdd-4861-862e-6357c89a58b2)|![Simulator Screenshot - iPhone 14 Pro - 2023-05-24 at 17 59 38](https://github.com/GetStream/stream-chat-swift/assets/472467/80b4964d-43f7-4124-8de8-815024fe18da)|

### 🧪 Manual Testing Notes

- Make sure that `isVoiceRecordingEnabled` is set to `true`
- Head into any Channel and tap on the mic button on the bottom right corner to begin recording.

### ☑️ Contributor Checklist


### 🎁 Meme

![gif](https://media4.giphy.com/media/Vnattpx6P6oDohAjhB/giphy.gif?cid=ecf05e47g8ofu5fl8aea49026bx8tmvtc0a68ebhzy1z3vpm&ep=v1_gifs_search&rid=giphy.gif&ct=g)
